src/Avalonia.FuncUI/DSL/Panels/Grid.fs

```diff
@@ -1,5 +1,67 @@
 namespace Avalonia.FuncUI.DSL
 
+[<AutoOpen>]
+module ColumnDefinition =  
+    open Avalonia.Controls
+    
+    [<RequireQualifiedAccess>]
+    type ColumnWidth =
+    /// Column is auto-sized to fit it's contents
+    | Auto
+    /// Column is sized in device independent pixels
+    | Pixel of float
+    /// Column is sized as a weighted proportion of available space
+    | Star of float
+
+    type ColumnDefinition with
+
+        /// <summary>
+        /// Define the properties of a Grid column
+        /// </summary>
+        /// <param name="width">Specify column width as automatic, proportional or in pixels</param>
+        /// <param name="minWidth">Optional minimum column width in pixels</param>
+        /// <param name="maxWidth">Optional maximum column width in pixels</param>
+        static member create(width: ColumnWidth, ?minWidth: float, ?maxWidth: float) =
+            let columnDefinition = ColumnDefinition()
+            match width with
+            | ColumnWidth.Auto -> columnDefinition.Width <- GridLength(0.0, GridUnitType.Auto)
+            | ColumnWidth.Pixel width -> columnDefinition.Width <- GridLength(width, GridUnitType.Pixel)
+            | ColumnWidth.Star proportionalWidth -> columnDefinition.Width <- GridLength(proportionalWidth, GridUnitType.Star)
+            minWidth |> Option.iter (fun minW -> columnDefinition.MinWidth <- minW)
+            maxWidth |> Option.iter (fun maxW -> columnDefinition.MaxWidth <- maxW)
+            columnDefinition
+
+[<AutoOpen>]
+module RowDefinition =  
+    open Avalonia.Controls
+
+    [<RequireQualifiedAccess>]
+    type RowHeight =
```

Comment:
Can we make this DU a `[<Struct>]` ?

Reply:
Yes can't think of any drawbacks and probably better as a value type.